src/core/ContractCallStack.ts

```diff
@@ -1,13 +1,14 @@
-import { mapReplacer } from '../utils/utils';
 import { InteractionData } from './modules/impl/HandlerExecutorFactory';
+import { randomUUID } from 'crypto';
 
 export class ContractCallStack {
-  readonly interactions: Map<string, InteractionCall> = new Map();
+  readonly interactions: { [key: string]: InteractionCall } = {};
```

Comment:
switching to a "normal" object, as the way how Map are serialized by the mapSerializer really add to the brain fuck.

Reply:
also, we need to create some graphical tool, that would read a call stack and would allow to navigate easilly in the tree structure.

Example call stack from the "deposit" (which makes an internal write - calls "claim" function on a token)
```
{
  "contractTxId": "-WjWKBO-ELt15JiJmiIEcU82YfiA1Pnd-KaAsm_XxLg",
  "depth": 1,
  "label": "",
  "id": "7c80ae34-fe2b-40e1-8176-bc7b9a8a3a3e",
  "interactions": {
    "vLIwBylVToAUT70uDW9rkH1q3beDWl4yAgA4MM5mqRs": {
      "interactionInput": {
        "txId": "vLIwBylVToAUT70uDW9rkH1q3beDWl4yAgA4MM5mqRs",
        "sortKey": "000000000004,0000000000000,263ceea54c7abb6701cd16bfcc06511eea850f2aa1bb47833400c7cb84f20357",
        "blockHeight": 4,
        "blockTimestamp": 1663347032,
        "caller": "yM3Oqjo8SbXDv3LrMbHpJi59miOAhnitccULBVyN--s",
        "functionName": "deposit",
        "functionArguments": {
          "function": "deposit",
          "tokenId": "EcaPQQGyIQFhKnS39b6lsaxXJsKUu1UwEiKgbvMybQ4",
          "qty": 1,
          "txID": "7N_BPg7X1ZtmjLgOhPpjxJhoHxZg7WEKj1ciSKfmRRQ"
        },
        "foreignContractCalls": {
          "EcaPQQGyIQFhKnS39b6lsaxXJsKUu1UwEiKgbvMybQ4": {
            "contractTxId": "EcaPQQGyIQFhKnS39b6lsaxXJsKUu1UwEiKgbvMybQ4",
            "depth": 2,
            "label": "",
            "id": "f8ddb6ed-d3aa-4865-99b0-8597ef652623",
            "interactions": {
              "vLIwBylVToAUT70uDW9rkH1q3beDWl4yAgA4MM5mqRs": {
                "interactionInput": {
                  "txId": "vLIwBylVToAUT70uDW9rkH1q3beDWl4yAgA4MM5mqRs",
                  "sortKey": "000000000004,0000000000000,263ceea54c7abb6701cd16bfcc06511eea850f2aa1bb47833400c7cb84f20357",
                  "blockHeight": 4,
                  "blockTimestamp": 1663347032,
                  "caller": "-WjWKBO-ELt15JiJmiIEcU82YfiA1Pnd-KaAsm_XxLg",
                  "functionName": "claim",
                  "functionArguments": {
                    "function": "claim",
                    "txID": "7N_BPg7X1ZtmjLgOhPpjxJhoHxZg7WEKj1ciSKfmRRQ",
                    "qty": 1
                  },
                  "foreignContractCalls": {}
                },
                "interactionOutput": {
                  "cacheHit": false,
                  "executionTime": 0,
                  "valid": true
                }
              }
            }
          }
        }
      }
    }
  }
}
```